core/src/main/java/org/elasticsearch/action/admin/indices/shards/IndicesShardStoresResponse.java

```diff
@@ -198,7 +198,7 @@ public void writeTo(StreamOutput out) throws IOException {
         @Override
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             node.toXContent(builder, params);
-            builder.field(Fields.VERSION, version);
+            builder.field(Fields.VERSION, legacyVersion);
```

Comment:
I did not rename the field to keep backward compatibility. This means that v3.0.0 will still return the version field. In v4 we should get rid of the field altogether. wdyt?